tests/i18n/test_compilation.py

```diff
@@ -188,3 +188,49 @@ def test_app_locale_compiled(self):
         call_command('compilemessages', locale=[self.LOCALE], stdout=StringIO())
         self.assertTrue(os.path.exists(self.PROJECT_MO_FILE))
         self.assertTrue(os.path.exists(self.APP_MO_FILE))
+
+
+class IgnoreDirectoryCompilationTest(MessageCompilationTests):
+    # re-use the exclude directory since it contains some locale fixtures.
+    work_subdir = 'exclude'
+    MO_FILE = '%s/%s/LC_MESSAGES/django.mo'
+    CACHE_DIR = os.path.join('cache', 'locale')
+    NESTED_DIR = os.path.join('outdated', 'v1', 'locale')
+
+    def setUp(self):
+        super().setUp()
+        copytree('canned_locale', 'locale')
+        copytree('canned_locale', self.CACHE_DIR)
+        copytree('canned_locale', self.NESTED_DIR)
+
+    def all_exist(self, dir, langs):
+        results = []
+        for lang in langs:
+            exists = os.path.exists(self.MO_FILE % (dir, lang))
+            results.append(exists)
+        return all(results)
+
+    def none_exist(self, dir, langs):
+        results = []
+        for lang in langs:
+            not_exist = os.path.exists(self.MO_FILE % (dir, lang)) is False
+            results.append(not_exist)
+        return all(results)
+
+    def test_one_locale_dir_ignored(self):
+        call_command('compilemessages', ignore=['cache'], stdout=StringIO())
+        self.assertTrue(self.all_exist('locale', ['en', 'fr', 'it']))
+        self.assertTrue(self.none_exist(self.CACHE_DIR, ['en', 'fr', 'it']))
+        self.assertTrue(self.all_exist(self.NESTED_DIR, ['en', 'fr', 'it']))
```

Comment:
Given these methods are all wrapped in `assertTrue` calls they should probably be converted to `assert_all_exists` and `assert_none_exists` that perform the assertion themselves.

e.g.

```python
def assert_all_exists(self, dir, langs):
    self.assertTrue(all(
        os.path.exists(self.MO_FILE % (dir, lang)) for lang in langs
    ))